packages/flutter/lib/src/material/bottom_navigation_bar.dart

```diff
@@ -90,35 +99,168 @@ class BottomNavigationBar extends StatefulWidget {
   /// The index into [items] of the current active item.
   final int currentIndex;
 
-  /// Defines the layout and behavior of a [BottomNavigationBar].
+  /// Defines the layout and behavior of a [BottomNavigationBar].  See
+  /// documentation for [BottomNavigationBarType] for information on the meaning
+  /// of different types.
   final BottomNavigationBarType type;
 
   /// The color of the selected item when bottom navigation bar is
-  /// [BottomNavigationBarType.fixed].
+  /// [BottomNavigationBarType.fixed].  [fixedColor] must be null if the [type]
+  /// is not [BottomNavigationBarType.fixed].
   final Color fixedColor;
 
   /// The size of all of the [BottomNavigationBarItem] icons.
   ///
   /// This value is used to to configure the [IconTheme] for the navigation
-  /// bar. When a [BottomNavigationBarItem.icon] widget is not an [Icon] the widget
-  /// should configure itself to match the icon theme's size and color.
+  /// bar. When a [BottomNavigationBarItem.icon] widget is not an [Icon] the
+  /// widget should configure itself to match the icon theme's size and color.
```

Comment:
I'd probably put this documentation ("`When a [BottomNavigationBarItem.icon] widget is not an [Icon] the￼...`") on BottomNavigationBarItem.icon itself, and phrase it as "`This widget should be one that respects the ambient [IconTheme], e.g. [Icon] or [ImageIcon].`".

Reply:
How about this (on BottomNavigationBarItem.icon)?:

```dart
  /// Typically the icon is an [Icon] or an [ImageIcon] widget. If another type
  /// of widget is provided then it should configure itself to match the current
  /// [IconTheme] size and color.
```

Because that's what it already reads.  I simplified the docs here, and pointed to BottomNavigationBarItem.icon instead.

Reply:
That's ok but it makes it sound like Icon and ImageIcon don't have to configure themselves, which is what I was trying to avoid in mine. Don't worry about it though. :-)